### DISCUSSION TITLE
Re: Calling Steps from Step Definitions

### DISCUSSION THREAD

**Discussion by dejj** [cucumber/common#2172] [Q&A Everything Else]
Where can I find a backup of https://github.com/cucumber/cucumber/wiki/Calling-Steps-from-Step-Definitions ?

I am in want of functional abstraction in Gherkin.
Testers are copying multi-step setups across feature files, making them ugly to maintain even using keyword `Background`.
Organization prevents them from re-writing Gherkin steps as Java code.
So I implemented a plugin enabling recursive calls, no `GivenStep` [1], no `Define` [4].

I'd like to review the arguments made against recursive calls, cited in 2013 by Aslak Hellesoy in https://web.archive.org/web/20240523132930/https://groups.google.com/g/cukes/c/DzE_kGZx94I

- [1] https://web.archive.org/web/20090221025546/http://blog.josephwilk.net/ruby/cucumber-waves-goodbye-to-givenscenario.html
- [2] https://github.com/cucumber/cucumber/wiki/Calling-Steps-from-Step-Definitions
- [3] https://web.archive.org/web/20111007112857/https://www.aslakhellesoy.com/post/11055981222/the-training-wheels-came-off
- [4] https://web.archive.org/web/20130114063428/http://www.technophobia.com/blog/read/experiences-with-bdd-and-cucumber-part-ii-substeps-439
- [5] https://web.archive.org/web/20200909225136/https://github.com/cucumber/cucumber-jvm/issues/483

Unfortunately, link [2] went away before the archive indexed it in 2021 (https://web.archive.org/web/20210506174624/https://github.com/cucumber/cucumber/wiki/Calling-Steps-from-Step-Definitions).

Does anyone retain a copy of the wiki article?

**Comment by mpkorstanje** [comment]
The wiki is still there but was disabled when it got replaced with the docs site. I've made a copy of the underlying git repository here:  https://github.com/cucumber/common-wiki

You'll want to start looking the state of the repository before all pages were changed to redirect to the docs page.

E.g: https://github.com/cucumber/common-wiki/blob/c1c82c140a3aaa50657e731357c10a6601f52560/Calling-Steps-from-Step-Definitions.textile

**Comment by luke-hill** [comment]
Also see here: https://github.com/cucumber/cucumber-ruby/issues/1362
